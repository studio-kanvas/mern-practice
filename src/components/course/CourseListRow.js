import React, {PropTypes} from 'react';
import {Link} from 'react-router-dom';



const style = {
    cursor:'pointer'
};


export default function CourseListRow({course, handleDelete}) {
    return(
        <tr>
            <td><a href={course.watchHref} target="_blank">{course.title}</a></td>
            <td>{course.authorId}</td>
            <td>{course.category}</td>
            <td>{course.length}</td>
            <td><Link to={`/course/${course.id}`}><i className="fa fa-pencil" aria-hidden="true"/></Link></td>
            <td><i className="fa fa-trash-o" aria-hidden="true" style={style} value={course.id} onClick={handleDelete}/></td>
        </tr>
    );    
}
		



CourseListRow.propTypes = {
    course: PropTypes.object.isRequired,
    handleDelete: PropTypes.func.isRequired,
};
