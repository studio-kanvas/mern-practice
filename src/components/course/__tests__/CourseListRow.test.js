import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import CourseListRow from '../CourseListRow';


describe('CourseListRow.test.js', () => {

    it('renders without crashing', () => {
        const props = {
            course: { id: 1, title: 'Java Clean Code' },
            handleDelete: jest.fn()
        };
        
        const wrapper = shallow(<CourseListRow {...props}/>);

        expect(wrapper).toHaveLength(1);

        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });
});
