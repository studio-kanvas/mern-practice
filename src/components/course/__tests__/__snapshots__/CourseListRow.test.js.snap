exports[`CourseListRow.test.js renders without crashing 1`] = `
<tr>
  <td>
    <a
      target="_blank">
      Java Clean Code
    </a>
  </td>
  <td />
  <td />
  <td />
  <td>
    <Link
      replace={false}
      to="/course/1">
      <i
        aria-hidden="true"
        className="fa fa-pencil" />
    </Link>
  </td>
  <td>
    <i
      aria-hidden="true"
      className="fa fa-trash-o"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
      value={1} />
  </td>
</tr>
`;
